packages/@lwc/engine-core/src/framework/modules/static-parts.ts

```diff
@@ -0,0 +1,95 @@
+    const numParts = parts.length;
+
+    // Optimization given that, in most cases, there will be one part, and it's just the root
+    if (numParts === 1) {
+        const firstPart = parts[0];
+        if (firstPart.partId === 0) {
+            // 0 means the root node
+            firstPart.elm = root;
+            return;
+        }
+    }
+
+    const partIdsToParts = new Map<number, VStaticPart>();
+    for (const staticPart of parts) {
+        partIdsToParts.set(staticPart.partId, staticPart);
+    }
+
+    let numFoundParts = 0;
+    const { previousSibling, getLastChild } = renderer;
+    const stack = [root];
+    let partId = -1;
+
+    // Depth-first traversal. We assign a partId to each element, which is an integer based on traversal order.
+    while (stack.length > 0) {
+        const elm = ArrayShift.call(stack)!;
+        partId++;
+
+        const part = partIdsToParts.get(partId);
+        if (!isUndefined(part)) {
+            part.elm = elm;
+            if (++numFoundParts === numParts) {
+                return; // perf optimization - stop traversing once we've found everything we need
+            }
+        }
+
+        // For depth-first traversal, prepend to the stack in reverse order
+        // Note that we traverse using `*Child`/`*Sibling` rather than `children` because the browser uses a linked
+        // list under the hood to represent the DOM tree, so it's faster to do this than to create an underlying array
+        // by calling `children`.
+        let child = getLastChild(elm);
+        while (!isNull(child)) {
+            ArrayUnshift.call(stack, child);
+            child = previousSibling(child);
+        }
+    }
+
+    if (process.env.NODE_ENV !== 'production') {
+        assert.isTrue(
+            numFoundParts === numParts,
+            `Should have found all parts by now. Found ${numFoundParts}, needed ${numParts}.`
+        );
+    }
+}
+
+/**
+ * Given an array of static parts, do all the mounting required for these parts.
+ *
+ * @param root - the root element
+ * @param vnode - the parent VStatic
+ * @param renderer - the renderer to use
+ */
+export function applyStaticParts(root: Element, vnode: VStatic, renderer: RendererAPI): void {
+    // On the server, we don't support ref (because it relies on renderedCallback), nor do we
+    // support event listeners (no interactivity), so traversing parts makes no sense
+    if (!process.env.IS_BROWSER) {
+        return;
+    }
+    const { parts, owner } = vnode;
+    if (isUndefined(parts)) {
+        return;
+    }
+
+    traverseAndSetElements(root, parts, renderer); // this adds `part.elm` to each `part`
+
+    for (const part of parts) {
+        // Event listeners are only applied once when mounting, so they are allowed for static vnodes
+        applyEventListeners(part, renderer);
+        // Refs are allowed as well
+        applyRefs(part, owner);
+    }
```

Comment:
What if you add these two methods inside packages/@lwc/engine-core/src/framework/modules/static-parts.ts line 29? It already loops through parts once